controllers/articlesController.js

```diff
@@ -156,4 +159,84 @@ export default class ArticlesController {
       return Response.send(res, STATUS.BAD_REQUEST, error, false);
     }
   }
+
+  /**
+ * @description users can bookmark articles for reading later.
+ * @static
+ * @param  {object} req - The request object
+ * @param  {object} res - The response object
+ * @return {object} - It returns the request response object
+ */
+  static async bookmark(req, res) {
+    const { slug } = req.params;
+    const userId = req.user.id;
+    let articleId;
+
+    try {
+      const user = await User.findOne({
+        where: {
+          id: userId
+        }
+      });
+
+      if (!user) return Response.send(res, STATUS.UNATHORIZED, [], 'You have to signed up to bookmark an article', false);
+      const articleFound = await Article.findOne({
+        where: {
+          slug
+        }
+      });
+
+      if (!articleFound) return Response.send(res, STATUS.NOT_FOUND, [], 'This article does not exist', false);
+      articleId = articleFound.id;
+      const bookmarkExist = await Bookmark.findOne({
+        where: {
+          articleId,
+          userId,
+        }
+      });
+
+      if (bookmarkExist) {
+        await Bookmark.destroy(
+          {
+            where: {
+              articleId,
+              userId
+            }
+          }
+        );
+        return Response.send(res, STATUS.OK, [], 'successfully unbookmarked this article', true);
+      }
+
+      const newBookmark = await Bookmark.create(
+        { userId, articleId }
+      );
+      return Response.send(res, STATUS.CREATED, newBookmark, 'successfully bookmarked this article', true);
+    } catch (error) {
+      return Response.send(res, STATUS.SERVER_ERROR, error.message, 'sorry! something went wrong', false);
+    }
+  }
+
+  /**
+* @description users can get all bookmarked articles.
+* @static
+* @param  {object} req - The request object
+* @param  {object} res - The response object
+* @return {object} - It returns the request response object
+*/
+  static async getBookmarkedArticles(req, res) {
```

Comment:
#### Comment
- Good method name given to this one , be consistent 😄

Reply:
thank you